lib/rex/post/meterpreter/ui/console/command_dispatcher/kiwi.rb

```diff
@@ -643,6 +644,76 @@ def scrape_passwords(provider, method, args)
     return true
   end
 
+  def report_smb_cred(credential_core)
+    # Assemble the options hash for creating the Metasploit::Credential::Login object
+    login_data = {
+      core: credential_core,
+      status: Metasploit::Model::Login::Status::UNTRIED,
+      address: ::Rex::Socket.getaddress(client.sock.peerhost, true),
+      port: 445,
+      service_name: 'smb',
+      protocol: 'tcp',
+      workspace_id: shell.framework.db.workspace.id
+    }
+
+    shell.framework.db.create_credential_login(login_data)
+  end
+
+  def create_cred(domain, credential_data)
+    unless domain.blank?
+      credential_data[:realm_key] = Metasploit::Model::Realm::Key::ACTIVE_DIRECTORY_DOMAIN
+      credential_data[:realm_value] = domain
+    end
+    credential_core = shell.framework.db.create_credential(credential_data)
+
+    credential_core
+  end
+
+  def report_creds(type, rows)
+    user, domain, pass = rows
```

Comment:
@cn-kali-team Marking this as unresolved, as whilst you updated the definition as per @smcintyre-r7's suggestion, you didn't update all the locations where the `report_creds()` function is called. As it stands right now, your changes have effectively broken the code since you didn't also update all the locations where `report_creds()` is called. 

@smcintyre-r7 I believe this was done due to the way the code attempts to build up a table. I agree its not the prettiest but this may require reworking a few of the calls to `report_creds()` to get this to work. I'll take a look into this more on Monday.

Reply:
After further discussion for clarification, this code has been updated. Next commit should fix up the `report_creds()` calls to ensure the complete application of a solution to this issue.